### DISCUSSION TITLE
Provide draft schedule in REST API

### DISCUSSION THREAD

**Discussion by eriktews** [pretalx/pretalx#1319] [Ideas] [answered]
At the moment, it looks like the slot field in the REST API is always empty when a schedule hast been designed in the schedule editor, but no version has been released yet: https://docs.pretalx.org/api/resources/submissions.html

It would be great when it would be possible to dump the current draft (unreleased) schedule via the API so that a third party tool or script could do some checks before officially releasing the schedule and notifying the speakers.

Maybe an additional query parameter could be passed to the submissions endpoint so that the slot fields are populated with the draft slots and not the released ones.

**Comment by rixx** [comment]
You can find the current WIP schedule at /api/events/<slug>/schedules/wip/

**Comment by eriktews** [comment]
Unfortunately that doesn't work for me. I get a list of the submissions I scheduled in the wip schedule, but the slots are still empty and there is no start and end date and time.

However, I saw that the orga/event/<EVENT>/schedule/api/talks/ URL endpoint gives me what I need (the wip stard and end times), but the code is missing there. However you can extract it from the URL field.

**Reply by rixx** [reply]
The URL I mentioned lists the start and end times if you make an authenticated request – what's missing for you in this response?
![2022-06-06T22:32:29+02:00](https://user-images.githubusercontent.com/2657547/172243891-32813fe0-b101-4c58-9524-72ab74437a44.png).

However, the /talks API endpoint should use the scheduled (released, public) times, not the unreleased ones. If it doesn't do that, I'd consider that a bug.

**Reply by eriktews** [reply]
Your screenshot shows a populated slot. When I run:

`curl -H "Authorization: Token YYYYYYYYYYYYYYYYYYYYYYY" https://pretalx-host/api/events/my-event/schedules/wip/?limit=1000`

Then I just get `"slot": null,`

When I use the other URL: `https://pretalx-host/orga/event/my-event/schedule/api/talks/`, then there is no slot, but populated `start` and `end` fields.

There is no released schedule yet.

**Comment by eriktews** [comment]
Hi, I think I may have found the solution. The **/schedules/wip/** endpoint may just list the confirmed submissions. The unofficial **/schedule/api/talks/** endpoint seems to include all submissions.

Could that be true and when so, should the schedules api maybe accept something like a state parameter, where you could also specify something like **state=all** to get the none confirmed ones as well?

**Comment by eriktews** [comment]
I'm sorry, but it looks like this wasn't the issue, there are still some confirmed submissions missing there.

**Reply by rixx** [reply]
If confirmed proposals (not pending confirmed!) are missing in the /submissions?state=confirmed endpoint, that's definitely a but and I'd need some kind of reproduction for that. The schedules endpoint otoh contains all talks *visible* in the current schedule, which isn't the same as "confirmed", so that would be expected behaviour.

**Reply by eriktews** [reply]
There don't seem to be any submissions missing in the /submissions endpoint.

My issue was that in the **/schedules/wip/** endpoint, some (confirmed) submissions that are already scheduled in pretalx (visible in the web-ui and in the unofficial /orga/event/my-event/schedule/api/talks/ endpoint) but not yet included in the released schedule don't have a slot there. So is this expected behaviour and when so, is there any official endpoint that would provide me with the draft schedule that is visible in the web-ui when drafting a schedule?

**Reply by rixx** [reply]
There is no API like that. I recently obtained funding for reworking the pretalx APIs entirely (including making them writable), so I don't expect this to change soon, and rather this endpoint to become deprecated in favour of a new API. (There's a reason that the /schedules endpoints never were documented.)

**Comment by eriktews** [comment]
So in a nutshell, it's probably best to use the /orga/ endpoint for a while and once the API becomes available, switch to it.

**Comment by rixx** [comment, marked as answer]
New API is available! The WIP schedule is available at both its ID, and at /api/events/<event>/schedules/wip/.
